homeassistant/components/switch/mysensors.py

```diff
@@ -0,0 +1,164 @@
+"""
+homeassistant.components.switch.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for MySensors switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.mysensors.html
+"""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    STATE_ON, STATE_OFF)
+
+import homeassistant.components.mysensors as mysensors
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = []
+
+ADD_DEVICES = None
+S_TYPES = None
+V_TYPES = None
+
+
+@mysensors.mysensors_update
+def sensor_update(gateway, port, devices, nid):
+    """Internal callback for sensor updates."""
+    return (S_TYPES, V_TYPES, MySensorsSwitch, ADD_DEVICES)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the mysensors platform for switches."""
+    # Define the S_TYPES and V_TYPES that the platform should handle as states.
+    global ADD_DEVICES
```

Comment:
Don't do this. This should stay encapsulated inside the `setup_platform` method.